lib/ui/painting.dart

```diff
@@ -1316,6 +1316,26 @@ Future<Null> _decodeImageFromListAsync(Uint8List list, ImageDecoderCallback call
   callback(frameInfo.image);
 }
 
+/// The encoding formats supported by the [encodeImage].
+enum EncodedImageFormat {
+  // Be conservative with the formats we expose. We can increase the list in future but reducing
+  // it is more difficult.
+  JPEG,
+  PNG,
+  WEBP,
+}
```

Comment:
these will need documentation, ideally including a pointer to the relevant specification for each one.

Reply:
Also for each one we should say what the `quality` argument does.

Reply:
These map to Skia values. Skia's does not document the supported format in much details. I was not able to even find link to relevant specification from them. So I have not added any spec links since I was not confident. If you have authoritative links I am happy to add.

Here is the Skia [documentation](https://skia.org/user/api/SkImage_Reference#SkImage_encodeToData_quality
) on quality. It has no details on exact impact of quality for each format beyond "it is format specific and may be ignored". 

My tests using the demo in the above link confirms that the quality actually works for this three
formats. So I have updated the wording to reflect this with more detail on low vs high.